builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -421,9 +421,16 @@ func resourceVSphereVirtualMachine() *schema.Resource {
 							Default:  "scsi",
 							ValidateFunc: func(v interface{}, k string) (ws []string, errors []error) {
 								value := v.(string)
-								if value != "scsi" && value != "ide" {
```

Comment:
bit of a moot point since it's merged already, but...  the use of a map may improve lookup performance, but for a small set like this it's probably negligible and it wouldn't matter much unless the operation was frequent.  Also, there's value in having DiskControllerTypes as an exported list for users of the module.  Turning the list in to a map in that case would be detrimental as DiskControllerTypes would have to be iterated as a map given there's no key extraction facility for maps in go.